chemprop/nn/metrics.py

```diff
@@ -144,33 +148,57 @@ def _calc_unreduced_loss(self, preds, targets, mask, *args) -> Tensor:
         return F.auroc(preds[mask], targets[mask].long(), task=self.task)
 
     def extra_repr(self) -> str:
-        return f"task={self.task}"
+        return f"task='{self.task}'"
 
 
 @MetricRegistry.register("prc")
 class AUPRCMetric(Metric):
     minimize = False
 
+    def __init__(self, task: str, **kwargs):
+        super().__init__(**kwargs)
+        self.task = task
+
     def forward(self, preds: Tensor, targets: Tensor, *args, **kwargs):
-        p, r, _ = F.precision_recall(preds, targets.long())
+        p, r, _ = F.precision_recall_curve(preds, targets.long(), task=self.task)
+        return auc(r, p)
```

Comment:
torchmetrics uses `torch.trapz(p, r) * -1` to calculate the auc.
See [here](https://github.com/Lightning-AI/torchmetrics/blob/be3970810a38753bfe53b8b162743de63bd6b179/src/torchmetrics/classification/precision_recall_curve.py#L219) and [here](https://github.com/Lightning-AI/torchmetrics/blob/master/src/torchmetrics/utilities/compute.py#L88)